code/modules/paperwork/paperbin.dm

```diff
@@ -60,38 +91,29 @@
 		to_chat(user, "<span class='notice'>You take [P] out of \the [src].</span>")
 		bin_pen = null
 		update_appearance()
-	else if(total_paper >= 1)
-		total_paper--
+	else if(LAZYLEN(papers))
+		var/obj/item/paper/top_paper = papers[LAZYLEN(papers)]
```

Comment:
```suggestion
		var/obj/item/paper/top_paper = papers[papers.len]
```

It is guaranteed at this point that there are papers, using LAZYLEN here implies it could be 0.